src/crypto/tls/cfkem.go

```diff
@@ -0,0 +1,88 @@
+// Glue to add Circl's (post-quantum) hybrid KEMs to quic-go.
+//
+// To enable set CurvePreferences with the desired scheme as the first element:
+//
+//   import (
+//      "github.com/cloudflare/circl/kem/tls"
+//      "github.com/cloudflare/circl/kem/hybrid"
+//
+//          [...]
+//
+//   config.CurvePreferences = []tls.CurveID{
+//      hybrid.Kyber512X25519().(tls.TLSScheme).TLSCurveID(),
+//      qtls.X25519,
+//      qtls.P256,
+//   }
+
+package tls
+
+import (
+	"io"
+
+	"circl/kem"
+	"circl/kem/hybrid"
+)
+
+// Either ecdheParameters or kem.PrivateKey
+type clientKeySharePrivate interface{}
+
+var (
+	kyber512X25519CurveID = CurveID(0xFF01)
```

Comment:
Is this codepoint defined in a draft, e.g., Stebila's hybrid KEX draft? If so we should point to it.

Reply:
Are 0xFF01 and 0xFFF02 arbitrary chosen? There is a conflict with existing parameters https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-8
```
65281 	arbitrary_explicit_prime_curves 	Y 	N 	[[RFC8422](https://www.iana.org/go/rfc8422)] 	
65282 	arbitrary_explicit_char2_curves 	Y 	N 	[[RFC8422](https://www.iana.org/go/rfc8422)]
```
These are `deprecated(0xFF01..0xFF02),` in https://www.rfc-editor.org/rfc/rfc8422.html#page-12

Maybe we can pick something from the "65024-65279  Reserved for Private Use" range?